server/src/main/java/org/jboss/as/server/BootstrapListener.java

```diff
@@ -103,11 +116,23 @@ private void done(final long bootstrapTime, final StabilityStatistics statistics
         final int passive = statistics.getPassiveCount();
         final int problem = statistics.getProblemsCount();
         final int started = statistics.getStartedCount();
+
+        String message;
+        if (configFiles == null || configFiles.isEmpty()) {
+            message = "";
+        } else if (configFiles.size() == 2) {
+            message = ServerLogger.AS_ROOT_LOGGER.hostConfigFilesInUse(configFiles.get(0), configFiles.get(1));
+        } else if (configFiles.size() == 1 && isHostController) {
+            message = ServerLogger.AS_ROOT_LOGGER.hostConfigFileInUse(configFiles.get(0));
+        } else {
+            message = ServerLogger.AS_ROOT_LOGGER.serverConfigFileInUse(configFiles.get(0));
+        }
```

Comment:
Hi @khermano , if you don't have a strong argument for doing this, I suggest reverting the changes that force you to add this logic and importing the `ServerLogger` into `DomainModelControllerService.java`. It adds more complexity and now you have related logic split into two areas, one in the `BootstrapListener.java` and one into `ServerService.java` and `DomainModelControllerService.java` (// Die! part). Notice these two classes don't need any logic to detect the different cases, you don't need to check if you are managing the Host Controller vs Server case. All of this is implicitly managed by the location of the code in these classes (`ServerService.java` and `DomainModelControllerService.java`).

However, after moving these cases into  `BootstrapListener.java` we are missing this status provided by the old location and you now need to deal with all the cases inside `BootstrapListener.java`. I don't get the motivation to do it, but share your argument if you think it can be justified.
 
I think we should stick to what we have been discussing: just split the generation of the message into two phases, one phase generates the `generateBootStatistics()`, and the other prints it on `postBoot()` without altering any behavioral change that could affect to the order of when things are done regarding when the process status change to RUNNING.

Reply:
Hi @yersan,
Yes, I agree it added unwanted complexity and use of ServerLogger in DomainModelControllerService, which is undesirable. I was trying to move all the messages creating responsibility (both boot statistics and configFilesInUse) into BootstrapListener, which had unwanted consequences, so I am splitting it into two parts.
I reverted these changes such that the configFilesInUse is created in ServerService/ DomainModelControllerService as before. Then I send it to the method BootstrapListener.printBootStatisticsMessage, where I combine the boot statistics message into one final one. 
What do you think about that?